core/storage/blog_statistics/gae_models.py

```diff
@@ -0,0 +1,999 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing statsistics data related to blog posts."""
+
+from __future__ import annotations
+
+import datetime
+
+from core import utils
+from core.platform import models
+
+from typing import Dict, List
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+
+(base_models, ) = models.Registry.import_models([models.Names.BASE_MODEL])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class BlogPostViewedEventLogEntryModel(base_models.BaseModel):
```

Comment:
For this and other classes please explain in the docstring how the model ID is generated.

Reply:
Added.